chat_api_sample/src/main/java/com/zopim/sample/chatapi/chat/ChatPresenter.java

```diff
@@ -0,0 +1,91 @@
+package com.zopim.sample.chatapi.chat;
+
+import android.support.v4.app.FragmentActivity;
+
+import com.zendesk.belvedere.BelvedereResult;
+import com.zendesk.util.CollectionUtils;
+import com.zopim.android.sdk.model.ChatLog;
+import com.zopim.android.sdk.model.Connection;
+
+import java.util.List;
+import java.util.Map;
+
+class ChatPresenter implements ChatMvp.Presenter {
+
+    private final ChatMvp.Model model;
+    private final ChatMvp.View view;
+
+    // keep a strong reference
+    @SuppressWarnings("FieldCanBeLocal")
+    private ChatMvp.Model.ChatListener chatListener;
+
+    ChatPresenter(final ChatMvp.Model model, final ChatMvp.View view) {
+        this.model = model;
+        this.view = view;
+    }
+
+    @Override
+    public void install(FragmentActivity fragmentActivity) {
+        chatListener = new MyChatListener();
+        model.registerChatListener(chatListener);
+
+        view.initChatLog(fragmentActivity);
+        view.setInputEnabled(false);
+        view.showLoading(true);
+    }
+
+    @Override
+    public void sendMessage(final String message) {
+        model.sendMessage(message);
+    }
+
+    @Override
+    public void sendFile(final List<BelvedereResult> belvedereResult) {
```

Comment:
Belvedere returns a `List`, but the user is only allowed to choose one a single file. 
I decided to put the logic for extracting the right item into the presenter to make it easier to test and stuff.  

https://github.com/zendesk/android_sdk_demo_apps/blob/schlan/BROS-111_chat_api_sample_app/chat_api_sample/src/main/java/com/zopim/sample/chatapi/chat/ChatActivity.java#L51